mozilla_django_oidc_db/plugins.py

```diff
@@ -49,48 +45,71 @@ def validate_settings(self) -> None:
         ...
 
     @abstractmethod
-    def filter_users_by_claims(self, claims: JSONObject) -> UserManager[AbstractUser]:
-        """Return all users matching the specified subject."""
-        ...
+    def handle_callback(self, request: HttpRequest) -> HttpResponse:
+        """Return an HttpResponse based on the callback view specified on the plugin.
 
-    @abstractmethod
-    def create_user(self, claims: JSONObject) -> AbstractUser | AnonymousUser:
-        """Create an authenticated user.
+        .. code:: python
 
-        This returns either a User if we need to have a new user created.
-        Otherwise, for application that don't need to create a new user whenever there is a login,
-        an AnonymousUser is returned.
-        """
-        ...
+           def view(self, request: HttpRequest) -> HttpResponse:
+               return admin_callback_view(request)
 
-    @abstractmethod
-    def update_user(self, user: AbstractUser, claims: JSONObject) -> AbstractUser:
-        """
-        Update existing user with new claims, if necessary save, and return user.
         """
         ...
 
     @abstractmethod
-    def handle_callback(self, request: HttpRequest) -> HttpResponse:
-        """Return an HttpResponse based on the callback view specified on the plugin.
+    def get_extra_params(self, request: HttpRequest, extra_params: dict) -> dict: ...
```

Comment:
can you make the `dict` types more specific? now it's essentially `dict[Any, Any]`

I think this should be `dict[str, str]` but I'm not sure